File: src-tauri/src/commands/network.rs
use crate::network::proxy::ProxyConfig;

/// Set the proxy configuration
/// This applies the proxy settings to environment variables that reqwest/kube-rs will use
#[tauri::command]
pub fn set_proxy_config(
    proxy_type: String,
    host: String,
    port: u16,
    username: String,
    password: String,
) -> Result<(), String> {
    let config = ProxyConfig {
        proxy_type,
        host,
        port,
        username,
        password,
    };

    config.apply_to_env();
    Ok(())
}

/// Get the current proxy configuration from environment
#[tauri::command]
pub fn get_proxy_config() -> ProxyConfig {
    crate::network::proxy::detect_system_proxy().unwrap_or_default()
}

File: src/lib/tauri/commands/network.ts
import { invoke } from "./core";

// Network/Proxy commands
export interface ProxyConfig {
  proxyType: string;
  host: string;
  port: number;
  username: string;
  password: string;
}

export async function setProxyConfig(
  proxyType: string,
  host: string,
  port: number,
  username: string,
  password: string
): Promise<void> {
  return invoke("set_proxy_config", {
    proxyType,
    host,
    port,
    username,
    password,
  });
}

export async function getProxyConfig(): Promise<ProxyConfig> {
  return invoke<ProxyConfig>("get_proxy_config");
}

/** Apply proxy settings from the persisted UI settings shape. */
export async function applyProxyFromSettings(settings: {
  proxyType: string;
  proxyHost: string;
  proxyPort: number;
  proxyUsername: string;
  proxyPassword: string;
}): Promise<void> {
  return setProxyConfig(
    settings.proxyType,
    settings.proxyHost,
    settings.proxyPort,
    settings.proxyUsername,
    settings.proxyPassword
  );
}
